persistence/sql/persister_consent.go

```diff
@@ -214,146 +211,186 @@ func (p *Persister) GetConsentRequest(ctx context.Context, challenge string) (*c
 		return nil, err
 	}
 
+	// We need to overwrite the ID with the encoded flow (challenge) so that the client is not confused.
+	f.ConsentChallengeID = sqlxx.NullString(challenge)
+
 	return f.GetConsentRequest(), nil
 }
 
-func (p *Persister) CreateLoginRequest(ctx context.Context, req *consent.LoginRequest) error {
+func (p *Persister) CreateLoginRequest(ctx context.Context, req *flow.LoginRequest) (*flow.Flow, error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.CreateLoginRequest")
 	defer span.End()
 
 	f := flow.NewFlow(req)
-	return sqlcon.HandleError(p.CreateWithNetwork(ctx, f))
+	nid := p.NetworkID(ctx)
+	if nid == uuid.Nil {
+		return nil, errorsx.WithStack(x.ErrNotFound)
+	}
+	f.NID = nid
+
+	return f, nil
 }
 
 func (p *Persister) GetFlow(ctx context.Context, loginChallenge string) (*flow.Flow, error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.GetFlow")
 	defer span.End()
 
 	var f flow.Flow
-	return &f, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		if err := p.QueryWithNetwork(ctx).Where("login_challenge = ?", loginChallenge).First(&f); err != nil {
-			if errors.Is(err, sql.ErrNoRows) {
-				return errorsx.WithStack(x.ErrNotFound)
-			}
-			return sqlcon.HandleError(err)
+	if err := p.QueryWithNetwork(ctx).Where("login_challenge = ?", loginChallenge).First(&f); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, errorsx.WithStack(x.ErrNotFound)
 		}
-
-		return nil
-	})
+		return nil, sqlcon.HandleError(err)
+	}
+	return &f, nil
 }
 
-func (p *Persister) GetLoginRequest(ctx context.Context, loginChallenge string) (*consent.LoginRequest, error) {
+func (p *Persister) GetLoginRequest(ctx context.Context, loginChallenge string) (*flow.LoginRequest, error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.GetLoginRequest")
 	defer span.End()
 
-	var lr *consent.LoginRequest
-	return lr, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		var f flow.Flow
-		if err := p.QueryWithNetwork(ctx).Where("login_challenge = ?", loginChallenge).First(&f); err != nil {
-			if errors.Is(err, sql.ErrNoRows) {
-				return errorsx.WithStack(x.ErrNotFound)
-			}
-			return sqlcon.HandleError(err)
-		}
-		lr = f.GetLoginRequest()
+	f, err := flowctx.Decode[flow.Flow](ctx, p.r.FlowCipher(), loginChallenge, flowctx.AsLoginChallenge)
+	if err != nil {
+		return nil, errorsx.WithStack(x.ErrNotFound.WithWrap(err))
+	}
+	if f.NID != p.NetworkID(ctx) {
+		return nil, errorsx.WithStack(x.ErrNotFound)
+	}
+	if f.RequestedAt.Add(p.config.ConsentRequestMaxAge(ctx)).Before(time.Now()) {
+		return nil, errorsx.WithStack(fosite.ErrRequestUnauthorized.WithHint("The login request has expired, please try again."))
+	}
+	lr := f.GetLoginRequest()
+	lr.ID = loginChallenge
```

Comment:
I assume thi sis done for the same reason like in `GetConsentRequest` but can you explain why it is needed?